### ISSUE TITLE
No documented resource on the Event bridge event triggered when a Chime Media Concatenate Pipeline is complete

### ISSUE THREAD

**Issue by Demi-ob** [aws/aws-sdk-js#4485]
### Describe the issue

What event bridge event is triggered when a pipeline concatenation is complete? I can't find any docs on it. 
I am working on implementing the AWS chime pipeline concatenation to merge videos/audio recorded from a chime session.
What Event bridge event is triggered when the concatenation is completed successfully? I don't see anything in the docs on it

### Links

https://docs.aws.amazon.com/chime-sdk/latest/dg/create-concat-pipe.html

**Comment by Demi-ob** [comment]
Any update here?

**Comment by ajredniwja** [comment]
Hey @Demi-ob thanks for opening this issue, I am not sure how you are consuming this SDK. The dos you have mentioned is of the chime-sdk. 

Usually you would trigger a custom event using the SDK and incorporate the event triggering to your pipeline.

If you are using the chime-sdk in JS which is different from aws-sdk-js, depending on the usage, I would ask the specific questions at https://github.com/aws-samples/amazon-chime-sdk you will also be able to find a couple of examples there.

**Comment by Demi-ob** [comment]
The docs I referenced is making use of `aws-sdk` in nodejs to create a MediaConcatenatePipeline on the backend.
e.g 
```
const createdMediaPipelineConcat = await chimeWithMediaPipeline
    .createMediaConcatenationPipeline(mediaConcatenatePipelineParams)
    .promise();
```


Everything else works fine, however there is no documentation anywhere on the Event Bridge event that gets triggered when a MediaCapturePipeline is completed (or any state change). 
Therefore I have no way to know that a MediaCapturePipeline is complete so I can update the record state in my own database.


For the MediaCapturePipeline (Not to be mistaken with MediaConcatenatePipeline), 
i.e
```
const createdMediaPipeline = await chimeWithMediaPipeline
    .createMediaCapturePipeline(mediaCapturePipelineParams)
    .promise();

```
I have an EventBridge listener that that enqueues an SQS on event or state change i.e 
 ` "chime:MediaPipelineInProgress"  "chime:MediaPipelineDeleted" "chime:MediaPipelinePermanentFailure"`
 Then I have a listener in my backend application on the SQS so I can process the different state changes.
 
 I could find any resource for this on `createMediaConcatenationPipeline` in any documentation or even in the EventBridge templates



Basically How do i know that a MediaConcatenationPipeline is complete, failed or state change?

**Comment by aBurmeseDev** [comment]
Hi @Demi-ob - apologies for the delay response. 

> I could find any resource for this on createMediaConcatenationPipeline in any documentation or even in the EventBridge templates

This service docs has infomation for `createMediaConcatenationPipeline` :https://docs.aws.amazon.com/chime-sdk/latest/APIReference/API_media-pipelines-chime_CreateMediaConcatenationPipeline.html


>Basically How do i know that a MediaConcatenationPipeline is complete, failed or state change?

As far as I'm aware, Chime SDK does not currently provide any specific EventBridge events or notifications for the completion or state changes of a MediaCapturePipeline. But we'll have to reach out to service team to confirm. You can reach service teams directly via [AWS Support](https://aws.amazon.com/contact-us/).

However, you can work around this limitation by using the `getMediaCapturePipeline` API method to periodically check the status of the `MediaCapturePipeline`. You can set up a scheduled task or a cron job to call this API at regular intervals and check the Status property of the `MediaCapturePipeline` response.

```js
async function checkMediaCapturePipelineStatus(pipelineId) {
  const response = await chime.getMediaCapturePipeline({ MediaPipelineId: pipelineId }).promise();

  switch (response.MediaCapturePipeline.Status) {
    case 'Initializing':
      console.log('MediaCapturePipeline is initializing...');
      break;
    case 'InProgress':
      console.log('MediaCapturePipeline is in progress...');
      break;
    case 'Stopped':
      console.log('MediaCapturePipeline has stopped.');
      // Perform any necessary actions, such as updating your database
      break;
    case 'Failed':
      console.log('MediaCapturePipeline has failed.');
      // Handle the failure case
      break;
    default:
      console.log(`Unexpected MediaCapturePipeline status: ${response.MediaCapturePipeline.Status}`);
  }
}

// Call the function at regular intervals (e.g., every minute)
setInterval(async () => {
  await checkMediaCapturePipelineStatus('your-pipeline-id');
}, 60000);
```

Hope it helps!
Best,
John

**Comment by vakhooooo** [comment]
Guys, you can do the following to understand when the concatenation is done (I don’t know yet how to track failure scenarios). To track when the concatenation completes successfully, you can use S3 events.

For example, in the CreateMediaConcatenationPipeline API, there is a configuration option for Sinks:

`Sinks: [ 
  { 
    Type: "S3Bucket", // required
    S3BucketSinkConfiguration: { 
      Destination: "STRING_VALUE", // required
    },
  },
],
`

When the concatenated file appears in the destination S3 bucket, you can listen to the S3 put event. This event can act as a trigger to indicate that the concatenation pipeline has finished.
